cs-rin-ru-enhanced-mod.user.js

```diff
@@ -449,13 +463,29 @@ function hideScs() {
     }
 }
 
-function colorize(str) {
+function hexToRgb(hex) {
+    const r = parseInt(hex.substring(0, 2), 16);
+    const g = parseInt(hex.substring(2, 4), 16);
+    const b = parseInt(hex.substring(4, 6), 16);
+    return [r, g, b];
+}
+
+function colorize(str, bgColour) {
     let lstr = str.toLowerCase();
     let hash = 0;
     for (let i = 0; i < lstr.length; i++) {
         hash = lstr.charCodeAt(i) + ((hash << 5) - hash);
     }
-    const color = Math.floor(Math.abs((Math.sin(hash) * 10000) % 1 * 16777216)).toString(16);
+    let color = Math.floor(Math.abs((Math.sin(hash) * 10000) % 1 * 16777216)).toString(16);
+    let rgb = hexToRgb(color);
+    let matches = bgColour.match(/^rgb\((\d+),\s*(\d+),\s*(\d+)\)$/);
+    const bgRgb = [parseInt(matches[1]), parseInt(matches[2]), parseInt(matches[3])]
+    console.log(bgRgb)
+    while (Math.abs(rgb[0] + rgb[1] + rgb[2] - (bgRgb[0] + bgRgb[1] + bgRgb[2])) < 200) {
```

Comment:
I haven't decided on what to put for the threshold, right now I have it set to 200. Maybe something higher like 300 would make it look nicer on the light theme.